tdrs-backend/tdpservice/core/views.py

```diff
@@ -50,7 +51,7 @@ def write_logs(request):
             # i.e. for newly uploaded DataFiles.
 
             LogEntry.objects.log_action(
-                user_id=User.objects.get(username=data['user']).pk,
+                user_id=request.user.pk,
```

Comment:
By capturing the user from the request instead of the data we can open this up to the permissions `IsAuthenticated` while still ensuring that users cannot create logs for other users.